eigensync/src/client/document.rs

```diff
@@ -0,0 +1,156 @@
+//! Automerge document management for client
+
+use crate::types::{Result, ActorId};
+use automerge::{AutoCommit, ChangeHash};
+use serde_json::Value;
+use std::collections::HashMap;
+
+/// Manager for Automerge documents
+pub struct DocumentManager {
+    documents: HashMap<String, AutoCommit>,
```

Comment:
We probably want to use a more specific type as our document key. `pub struct DocumentId(pub String)` for example. This ensures we don't accidentally mistake a values that's not a documentId as an id. Any reason we wouldn't use `Uuid`?